ts-client/src/amm/index.ts

```diff
@@ -50,6 +54,20 @@ type Opt = {
   cluster: Cluster;
 };
 
+const getAllPoolState = async (poolMints: Array<PublicKey>, program: AmmProgram) => {
+  const poolStates = (await program.account.pool.fetchMultiple(poolMints)) as Array<PoolState>;
+  invariant(poolStates.length === poolMints.length, 'Some of the pool state not found');
+
+  return await Promise.all(
+    poolStates.map(async (poolState) => {
+      const account = await program.provider.connection.getTokenSupply(poolState.lpMint);
```

Comment:
we can getMultipleAccounts then get the lpMint only in 1 call

Reply:
you might need to use MintLayout to decode it if you want to extract lpMint from gMA

Reply:
hmm, but `poolState` is fetch from the program using `fetchMultiple`

Reply:
isn't `getTokenSupply` the same with using MintLayout decoding?